src/NUnitFramework/framework/TestContext.cs

```diff
@@ -417,6 +417,24 @@ public PropertyBagAdapter Properties
                 get { return _test.Arguments; }
             }
 
+            /// <summary>
+            /// The Type of the test.
+            /// Returns null if no fixture type is associated with it.
+            /// </summary>
+            public Type? TypeInfo
```

Comment:
I would rather expose this as `FixtureType` or `Type`. However, can you explain at a higher level why ClassName doesn't give you the information you need?